### PR TITLE
Add linter to pipeline

### PR DESCRIPTION
**Pull request by jmgq**
This also fixes/ignores existing linting errors.

### CODE REVIEW

File: src/app/components/pages/TeacherOrTutorRequest.tsx [hunk 1]
```diff
@@ -38,10 +38,10 @@ export const TeacherOrTutorRequest = ({ user }: { user: RegisteredUserDTO }) =>
                   or a tutor:
                 </p>
                 <ul>
-                  <li>A "teacher" must be directly associated with a school</li>
+                  <li>A &quot;teacher&quot; must be directly associated with a school</li>
```
Comment (hannah-whelan) [inline src/app/components/pages/TeacherOrTutorRequest.tsx:41]: This is much less readable, do we have to replace quotes with &quot; ?

**Comment by hannah-whelan** [comment]
E2E tests pass and manual testing appears fine

File: src/app/components/content/IsaacReorderQuestion.tsx [hunk 2]
```diff
@@ -174,7 +174,13 @@ const IsaacReorderQuestion = ({ doc, questionId, readonly }: IsaacQuestionProps<
                 >
                   {availableItems &&
                     availableItems.map((item, index) => (
-                      <ReorderDraggableItem item={item} index={index} inAvailableItems readonly={readonly} />
+                      <ReorderDraggableItem
+                        item={item}
+                        index={index}
+                        key={index}
```
[the remark this answers is not in the document]
Comment (jmgq) [reply src/app/components/content/IsaacReorderQuestion.tsx:180]: Yes, I could set the key to the item ID, but that can be undefined... I can always suppress this rule instead. I've done something similar in `LevelsFilter`, so I'm happy to simply ignore it there too.

File: src/app/components/pages/TeacherOrTutorRequest.tsx (hunk 1 above)
Comment (jmgq) [reply src/app/components/pages/TeacherOrTutorRequest.tsx:41]: I think it's a good rule, here's an explanation of why this is a thing: https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/no-unescaped-entities.md

**Review by hannah-whelan — APPROVED** [review]

**Pull request merged**